### PR TITLE
k8s-local-kind: kill old cluster if REUSE_CLUSTER is not set

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/iCmqi4N2/3622-investigate-functional-tests-issues

Make kind cluster to be deleted if no SCT_REUSE_CLUSTER is set

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster_k8s/mini_k8s.py [hunk 1]
```diff
@@ -616,4 +631,4 @@ def nodes_dest_ip(self) -> Optional[str]:
 
 
 class MonitorSetMinikube(MonitorSetGCE):
-    def install_scylla_manager(self, node, auth_token): ...
+    def install_scylla_manager(self, node): ...
```
Comment (vponomaryov) [inline sdcm/cluster_k8s/mini_k8s.py]: Is it pylint fix? Anyway, it doesn't look related to the goal of the commit.

File: sdcm/cluster_k8s/mini_k8s.py [hunk 2]
```diff
@@ -189,6 +188,9 @@ def start_k8s_software(self) -> None:
         """)
         self.host_node.remoter.run(f"sudo -E bash -cxe '{script}'")
 
+    def stop_k8s_software(self):
+        self.host_node.remoter.run('kind delete cluster', ignore_status=True)
```
Comment (vponomaryov) [inline sdcm/cluster_k8s/mini_k8s.py:191]: Shouldn't we make sure that cluster is deleted? If it fails to be deleted then it will keep computer resources used and we won't be able to create new cluster.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/cluster_k8s/mini_k8s.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/cluster_k8s/mini_k8s.py]: yes, it is pylint issue

File: sdcm/cluster_k8s/mini_k8s.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/cluster_k8s/mini_k8s.py:191]: kind can fail on cluster deletion by multiple reasons, not all of them means that cluster wasn't actually deleted.
It is safe to fail later on cluster creation

**Review by bentsi — APPROVED** [review]
